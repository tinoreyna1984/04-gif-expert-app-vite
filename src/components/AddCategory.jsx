import React from 'react'
import { useState } from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({addCategory}) => {

    const [value, setValue] = useState('');

    const handleChange = (e) => {
        setValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if(value.trim().length < 1) return;
        setValue('');
        addCategory(value.trim());
    }
  return (
    <form onSubmit={handleSubmit} aria-label='form'>
        <input placeholder='agregar gifs' onChange={handleChange} value={value} />
    </form>
  )
}

AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired
}
