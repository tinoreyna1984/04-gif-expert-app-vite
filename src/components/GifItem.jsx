import React from "react";
import PropTypes from "prop-types";

export const GifItem = ({image}) => {
  return (
    <div className="card">
      <img src={image.url} alt={image.title} />
    </div>
  );
};

GifItem.propTypes = {
  image: PropTypes.object.isRequired,
}
