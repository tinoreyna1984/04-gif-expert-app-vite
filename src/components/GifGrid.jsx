import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

import spinner from '../../assets/blocks-scale-spinner.svg';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <img src={spinner} alt="Loading..." />}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} image={image} />
        ))}
      </div>
    </>
  );
};

