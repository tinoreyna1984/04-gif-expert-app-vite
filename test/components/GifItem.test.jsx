import { fireEvent, render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const image = {
    url: "https://media1.giphy.com/media/3o7btLwXyvzqgXQq2U/giphy.gif",
    title: "Gif de prueba",
  };

  test("El objeto de imagen debe coincidir con el snapshot", () => {
    const { container } = render(<GifItem image={image} />);
    screen.debug();
    expect(container).toMatchSnapshot();
  });

  test("El objeto de imagen debe ingresar obligatoriamente", () => {
    render(<GifItem image={image} />);
    expect(screen.getByAltText("Gif de prueba")).toBeTruthy();
  });

  test("Los valores de src y alt deben ser los adecuados", () => {
    render(<GifItem image={image} />);
    const {src, alt} = screen.getByRole("img");
    expect(src).toBe(image.url);
    expect(alt).toBe(image.title);
  });
});
