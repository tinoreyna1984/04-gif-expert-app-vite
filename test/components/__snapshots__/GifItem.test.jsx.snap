// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem El objeto de imagen debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif de prueba"
      src="https://media1.giphy.com/media/3o7btLwXyvzqgXQq2U/giphy.gif"
    />
  </div>
</div>
`;
